zipkin-collector/kafka/src/main/java/zipkin2/collector/kafka/KafkaCollector.java

```diff
@@ -164,15 +172,19 @@ public CheckResult check() {
     try {
       CheckResult failure = kafkaWorkers.failure.get(); // check the kafka workers didn't quit
       if (failure != null) return failure;
+      if (adminClient == null) adminClient = AdminClient.create(properties);
```

Comment:
if the goal is to initialize (then later close) the admin client once, I think you will want to double check initialize with volatile ref like some other code in this repo

reason is close isnt guaranteed to be same thread as check

Reply:
Thanks for the advise, I'll check it.